src/github.com/matrix-org/dendrite/clientapi/sync/requestpool.go

```diff
@@ -112,7 +112,8 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request) util.JSONRespons
 }
 
 // OnNewEvent is called when a new event is received from the room server. Must only be
-// called from a single goroutine.
+// called from a single goroutine, or else the current position in the stream may be
+// set incorrectly as it is blindly clobbered.
```

Comment:
I think it would be clearer if you explicitly mentioned a race between goroutines here; "... single goroutine, to avoid races between updates which could set the current position in the stream incorrectly"